packages/components/src/components/inputs/Input/index.native.tsx

```diff
@@ -1,20 +1,156 @@
-/* eslint-disable no-unused-vars */
 import React from 'react';
-// import styled, { css } from 'styled-components/native';
-import PropTypes from 'prop-types';
-import { FeedbackType } from '../../../support/types';
+import styled, { css } from 'styled-components/native';
 
-interface InputProps {
+import { FONT_SIZE_NATIVE, FONT_WEIGHT, LINE_HEIGHT } from '../../../config/variables';
+import { getStateIcon } from '../../../utils/icons';
+import { getPrimaryColor } from '../../../utils/colors';
+import Icon from '../../Icon';
+import colors from '../../../config/colors';
+import { Omit, FeedbackType } from '../../../support/types';
+
+const Wrapper = styled.View<WrapperProps>``;
+
+const InputWrapper = styled.View`
+    flex: 1;
+    flex-direction: row;
+`;
+
+const TopLabel = styled.Text`
+    padding-bottom: 10px;
+    color: ${colors.TEXT_SECONDARY};
+`;
+
+const StyledInput = styled.TextInput<Props>`
+    width: 100%;
+    height: ${props => (props.height ? props.height : 40)};
+    padding: 0 ${props => (props.hasIcon ? '40px' : '12px')} 0 12px;
+
+    font-size: ${props => (props.isSmallText ? FONT_SIZE_NATIVE.SMALL : FONT_SIZE_NATIVE.BASE)};
+    font-weight: ${FONT_WEIGHT.MEDIUM};
+    color: ${props => (props.color ? props.color : colors.TEXT)};
+
+    border-radius: 2px;
+    
+    ${props =>
+        props.hasAddon &&
+        css`
+            border-top-right-radius: 0;
+            border-bottom-right-radius: 0;
+        `}
+
+    border: 1px solid ${props => props.border};
+
+    background-color: ${colors.WHITE};
+
+    ${props =>
+        props.tooltipAction &&
+        css`
+            z-index: 10001; /* bigger than modal container */
+            position: relative;
+        `};
+`;
+
+const StyledIcon = styled(Icon)`
+    position: absolute;
+    top: 12;
+    right: 15;
+    z-index: 10001;
+`;
+
+const BottomText = styled.Text<InputProps>`
+    margin-top: 10px;
+    font-size: ${FONT_SIZE_NATIVE.SMALL};
+    color: ${props => (props.color ? props.color : colors.TEXT_SECONDARY)};
+    z-index: 0;
+`;
+
+const Overlay = styled.View<InputProps>`
+    ${props =>
+        props.isPartiallyHidden &&
+        css`
+            bottom: 0;
+            border: 1px solid ${colors.DIVIDER};
+            border-radius: 2px;
+            position: absolute;
+            width: 100%;
+            height: 100%;
+            background-image: linear-gradient(
+                to right,
+                rgba(0, 0, 0, 0) 0%,
+                rgba(249, 249, 249, 1) 220px
+            );
+        `}
+`;
+
+const TooltipAction = styled.View`
+    display: ${(props: { action?: React.ReactNode }) => (props.action ? 'flex' : 'none')};
```

Comment:
it is better to define types in var declaration (`styled.View<{action?: React.ReactNode}>` or better yet extract it to separate var, eg `TooltipActionProps`) so it is always in one place

Reply:
fixed